### DISCUSSION TITLE
Privacy concerns, while signing up on Hoppscotch using Github

### DISCUSSION THREAD

**Discussion by NeelParihar** [hoppscotch/hoppscotch#1618] [General]
I was going through the https://hoppscotch.io web app, and when i was signing up using github it requested me to provide access to gists with both read and write access and also for secret gists, just wanted to know why it is asking for that particular permission and how Hoppscotch is using that information for a particular user.

_Following is the screenshot of screen which i had encountered while signing in using github._

<img width="631" alt="Screenshot 2021-04-26 at 7 29 39 PM" src="https://user-images.githubusercontent.com/51474217/116098166-7619bb80-a6c8-11eb-93a2-75fb30b2f6d2.png">

**Comment by liyasthomas** [comment]
Hi @NeelParihar thanks for showing interest to check out Hoppcotch.

We use `gist` access in two features:
1. Create secret gist of your personal collections + allow importing such gist URL to collections
2. Generate Markdown rendered API documentation

**1. Create secret gist of collections:**
Create and import collections with gist from Import / Export collections window.

<img width="1368" alt="Screenshot 2021-04-26 at 8 31 51 PM" src="https://user-images.githubusercontent.com/10395817/116104883-abd98880-a665-11eb-9b87-bebed058e963.png">

**2. Generate Markdown rendered API documentation**
Select collections and generate documentation

<img width="1368" alt="Screenshot 2021-04-26 at 8 33 24 PM" src="https://user-images.githubusercontent.com/10395817/116105508-3e7a2780-a666-11eb-9c80-601c9c22f839.png">

Generate gist

<img width="1368" alt="Screenshot 2021-04-26 at 8 36 34 PM" src="https://user-images.githubusercontent.com/10395817/116105790-8305c300-a666-11eb-8c5a-ea7027156d92.png">

Let me know if you've any other doubts.

**Reply by NeelParihar** [reply]
Thank you for the quick response @liyasthomas. That is really a nice feature, I didn't think about that. And also thank you for providing the information about it in such detail. I apologize if I have caused any inconvenience. 
 
Also nice to see that Hoppscotch automatically detects if I am signing in using the same email id which I earlier used to sign in from different OAuth providers.
